### PR TITLE
docs: add chainsaw capabilities to quick start guide

### LINKED ISSUE #961: [Docs] Improve the quick start guide
Our quick start guide is not representative enough of the chainsaw capabilities.
It should at least demonstrate assertion trees capabilities.

### PR DESCRIPTION
**Pull request by swastik959**
## Explanation
added chainsaw capabilities to quick start guide

## Related issue

Contributes to #961 

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/kyverno/kyverno/blob/main/CONTRIBUTING.md).
- [x] I have read the [PR documentation guide](https://github.com/kyverno/kyverno/blob/main/.github/pr_documentation.md) and followed the process including adding proof manifests to this PR.
- [x] This is a bug fix and I have added unit tests that prove my fix is effective.

### CODE REVIEW

**Comment by swastik959** [comment]
@eddycharly please review this pr also should I put the code snippets as well as elaborate assertion trees a bit more ?

**Comment by shubham-cmyk** [comment]
@swastik959 can you make the docs little more to the point and compact.

**Comment by swastik959** [comment]
@shubham-cmyk please have a look now

File: website/docs/quick-start.md [hunk 1]
```diff
@@ -146,3 +146,8 @@ Tests Summary...
 - Skipped tests 0
 Done.
 ```
+### Chainsaw Charaterstics 
+
+**Resource diff in assertion failures:** Chainsaw now offers detailed resource diffs upon assertion failures. This feature provides a clear view of discrepancies between expected and actual resource configurations, aiding testers in quickly identifying and resolving issues during debugging.
```
Comment (shubham-cmyk) [inline website/docs/quick-start.md:151]:
```suggestion
**Resource diff in assertion failures:** Chainsaw offers detailed resource diffs upon assertion failures. It provides a clear view of discrepancies between expected and actual resource.
```

**Review by shubham-cmyk — APPROVED** [review]

**Review by shubham-cmyk — APPROVED** [review]

**Review by eddycharly — COMMENTED** [review]
I would expect something more hands on than a bullet list.
Of course listing those features are better than nothing though.

**Comment by eddycharly** [comment]
@swastik959 i'm ok to merge this but it doesn't address the issue IMHO.

**Comment by swastik959** [comment]
@eddycharly I believe we can make a separate section for resource templating in the writing tests section

**Comment by eddycharly** [comment]
I'll merge it but remove the `Fixes` statement.

**Review by eddycharly — APPROVED** [review]

**Pull request merged**